workflows_core/api/api.py

```diff
@@ -257,5 +257,5 @@ def _update_workflow_metadata(self, job_id: str, metadata: Dict[str, Any]):
         return requests.post(
             url=self._base_url + f"/workflows/{job_id}/metadata",
             headers=self._headers,
-            json=dict(metadat=metadata),
+            json=dict(metadata=metadata),
```

Comment:
Why is there no test for this? This shouldn't have been a problem in the first place? Are you sure this works?